test/system/src/main/java/io/pravega/test/system/framework/kubernetes/K8sClient.java

```diff
@@ -614,7 +617,13 @@ public void deletePVC(String namespace, String name) {
                             String logFile = toFile + "-" + retryCount.incrementAndGet() + ".log";
                             Files.copy(logStream, Paths.get(logFile));
                             log.debug("Logs downloaded from pod {} to {}", podName, logFile);
+                            return null;
                         } catch (ApiException | IOException e) {
+                            if (e instanceof UnknownHostException) {
+                                // do not retry if an UnknownHostException is observed.
```

Comment:
Should we retry a bounded number of times? That would accommodate some transient failures, but not block the job indefinitely.

Reply:
The retry is  done a bounded number of times. I have updated the code and description of the PR.